_posts/getting-started/2024-03-04-using-the-cli-linux.md

```diff
@@ -0,0 +1,177 @@
+---
+layout: page
+tags: ["Getting-Started"]
+page_title: Using the OpenStack CLI (Linux)
+---
+
+# Using the OpenStack CLI (Linux)
+
+For Windows installations please see
+[Using the OpenStack CLI (Windows)]({{ '/articles/using-the-cli-windows' | relative_url }}).
+
+The OpenStack Command Line Interface (CLI) is a powerful tool for managing
+OpenStack resources. This guide will show you how to install the OpenStack CLI,
+log in to the OpenStack CLI, and use the OpenStack CLI to manage OpenStack
+resources.
+
+This guide makes use of the `clouds.yaml` file to store OpenStack credentials.
+You can also use environment variables to store your credentials using the
+`openrc` file. The `clouds.yaml` file is the recommended way to store your
+OpenStack credentials, as it is easier to manage and add multiple clouds
+(projects, regions etc.).
+
+---
+
+## Install the OpenStack CLI
+The installation of the OpenStack CLI is different for each Operating System 
+and distrubution. Below you will find the installation instructions for the
+most common Linux distributions.
+
+For Windows installations please see
+[Using the OpenStack CLI (Windows)]({{ '/articles/using-the-cli-windows' | relative_url }}).
+
+
+### Debian (10, 11, 12) | Ubuntu (20.04 | 22.04)
+To install the OpenStack CLI, run the following commands:
+
+**Step 1**  
+Update our package list to make sure we install the latest version of the
+OpenStack CLI.
+```bash
+sudo apt update
+```
+
+**Step 2**  
+Install the OpenStack CLI
+```bash
+sudo apt install python3-openstackclient
```

Comment:
I chose to use the apt install of the python3-openstackclient because when you use the pip install on Debian you retrieve errors because of the system managed python in Debian.
```
error: externally-managed-environment

× This environment is externally managed
╰─> To install Python packages system-wide, try apt install
    python3-xyz, where xyz is the package you are trying to
    install.
```
This can easily be overcome with using Virtual Environments but in my personal opinion I think it may make the process a bit more difficult then really required. You can also use the --break-system-packages flag but the flag itself says enough in my opinion to not use it, it might work today but I do not know what will happen in a month or even longer.

Although I do agree using the pip install would be the best because it installs the most recent versions I do think the apt install version for this tutorial is sufficient.

Reply:
Maybe make a remark, that the apt versions are not necessarily the latest versions, so that features will differ between OS versions, and provide an alternative?

Reply:
I will split the Ubuntu and Debian install because Ubuntu does not have the same issue. For Debian I will add an remark